tools/start-local-vm

```diff
@@ -0,0 +1,242 @@
+usage: ${0##*/} [--arch BUILDSYS_ARCH] [--variant BUILDSYS_VARIANT]
+                      [--vm-memory VM_MEMORY] [--vm-cpus VM_CPUS]
+                      [--inject-file LOCAL_PATH[:IMAGE_PATH]]...
+
+Launch a local virtual machine from a Bottlerocket image.
+
+Options:
+
+    --arch              architecture of the Bottlerocket image (must match the
+                        host architecture ($(uname -m)); may be omitted if the
+                        BUILDSYS_ARCH environment variable is set)
+    --variant           Bottlerocket variant to run (may be omitted if the
+                        BUILDSYS_VARIANT environment variable is set)
+    --vm-memory         amount of memory to assign to the VM; VM_MEMORY must be
+                        a valid QEMU memory specifier (default is ${vm_mem})
+    --vm-cpus           number of CPUs to spawn for VM (default is ${vm_cpus})
+    --force-extract     force recreation of the extracted Bottlerocket image,
+                        e.g. to incorporate new --inject-file options
+    --inject-file       adds a local file to the private partition of the
+                        Bottlerocket image before launching the virtual machine
+                        (may be given multiple times)
+    --help              shows this usage text
+
+The virtual machine's port 22 (SSH) will be exposed via the local port 2222,
+i.e. if the Bottlerocket admin container has been enabled via user-data, it can
+be reached by running
+
+    ssh -p 2222 ec2-user@localhost
+
+from the host.
+
+Usage example:
+
+    ${0##*/} --arch $(uname -m) --variant metal-dev --inject-file net.toml
+"
+
+    if [[ -n ${error} ]]; then
+        bail "${error}${newline}${newline}${usage}"
+    else
+        bail "${usage}"
+    fi
+}
+
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            -h|--help)
+                show_usage ;;
+            --arch)
+                shift; arch=$1 ;;
+            --variant)
+                shift; variant=$1 ;;
+            --vm-memory)
+                shift; vm_mem=$1 ;;
+            --vm-cpus)
+                shift; vm_cpus=$1 ;;
+            --force-extract)
+                force_extract=yes ;;
+            --inject-file)
+                shift; local file_spec=$1
+                if [[ ${file_spec} = *:* ]]; then
+                    local local_file=${file_spec%%:*}
+                    local image_file=${file_spec#*:}
+                else
+                    local local_file=${file_spec}
+                    local image_file=${file_spec##*/}
+                fi
+                extra_files[${local_file}]=${image_file}
+                ;;
+            *)
+                show_usage "unknown option '$1'" ;;
+        esac
+        shift
+    done
+
+    [[ -n ${arch} ]] || show_usage 'Architecture needs to be set via either --arch or BUILDSYS_ARCH.'
+    [[ -n ${variant} ]] || show_usage 'Variant needs to be set via either --variant or BUILDSYS_VARIANT.'
```

Comment:
I hadn't thought about integration into cargo-make yet but it is something to consider (perhaps after the script learns how to auto-generate some configuration such as `net.toml`). My original motivation for those indeed was convenience in case one has those exported anyway.